src/components/layout.ts

```diff
@@ -6,30 +6,68 @@ import {
   makeStyles,
 } from "@material-ui/core/styles";
 import { amber, lightBlue, grey } from "@material-ui/core/colors";
+import useMediaQuery from '@material-ui/core/useMediaQuery';
 
 import { rhythm, scale } from "../utils/typography";
 
 import Header from "./header";
 
 declare var __PATH_PREFIX__: string;
 
-const palette = {
+const paletteLight = {
   primary: {
     light: amber[500],
     main: amber[700],
     dark: amber[900],
     contrastText: "#ffffff",
   },
+  secondary: {
+    light: lightBlue[500],
+    main: lightBlue[700],
+    dark: lightBlue[900],
+    contrastText: "#ffffff",
+  },
+} as const;
+
+const paletteDark = {
+  type: "dark",
+  background: {
+    default: grey[900],
+    paper: grey[700],
+  },
+  text: {
+    primary: grey[100],
+  },
+  primary: {
+    light: amber[500],
+    main: amber[700],
+    dark: amber[900],
+    contrastText: "#dddddd",
+  },
   secondary: {
     light: lightBlue[100],
     main: lightBlue[300],
     dark: lightBlue[500],
-    contrastText: "#ffffff",
+    contrastText: "#dddddd",
   },
 } as const;
 
-const theme = createMuiTheme({
-  palette,
+const themeLight = createMuiTheme({
+  palette: paletteLight,
+  typography: {
```

Comment:
if the `typography` is the same and only `pallete` is changing, should we make it it's own var to be shared?